File: editor/en/[463]Island Perimeter.java

//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public int islandPerimeter(int[][] grid) {
        int rows = grid.length;
        int cols = grid[0].length;
        int counter = 0;

        for (int i = 0; i < rows; i++) {
            for (int j = 0; j < cols; j++) {
                if (grid[i][j] == 1) {
                    counter += 4;

                    if (i > 0 && grid[i - 1][j] == 1) {
                        counter -= 2;
                    }

                    if (j > 0 && grid[i][j - 1] == 1) {
                        counter -= 2;
                    }
                }
            }
        }

        return counter;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[463]Island Perimeter.py
# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        rows = len(grid)
        cols = len(grid[0])
        result = 0

        for i in range(rows):
            for j in range(cols):
                if grid[i][j] == 1:
                    result += 4

                    if i > 0 and grid[i - 1][j] == 1:
                        result -= 2
                    if j > 0 and grid[i][j - 1] == 1:
                        result -= 2

        return result
# leetcode submit region end(Prohibit modification and deletion)
